app/models/manageiq/providers/hawkular/middleware_manager/middleware_diagnostic_report.rb

```diff
@@ -0,0 +1,139 @@
+require 'timeout'
+
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::MiddlewareDiagnosticReport < ApplicationRecord
+    self.table_name = 'middleware_diagnostic_reports'
+
+    STATUS_QUEUED = 'Queued'.freeze
+    STATUS_RUNNING = 'Running'.freeze
+    STATUS_ERROR = 'Error'.freeze
+    STATUS_READY = 'Ready'.freeze
```

Comment:
Hmm... I have my doubts. Rails enums are mapped to integers in the database and most of the models I have seen, are using strings if they have statuses associated to them (probably to be verbose when querying directly the database). See, for example, [MiqQueue](https://github.com/ManageIQ/manageiq/blob/ef367b7e2e34750d3142494c42327c5c13da3dfb/app/models/miq_queue.rb#L75), [MiqTask](https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_task.rb#L3).

Although, if those models were written before rails supported enums, that could explain why they are coded like that.

Hmm... Schema PR is already merged, so the column is a string. I could try to see if enums work properly with a database string column to avoid having to place another PR to "fix" the column.